crates/vm/wasm/src/lib.rs

```diff
@@ -2,6 +2,7 @@ mod cache;
 mod environment;
 mod error;
 mod gas;
+mod gatekeeper;
```

Comment:
You should `pub use gatekeeper::*`, as with all other modules below.

Reply:
This is just my personal style. I like to use a "flat structure" as said in the coding guideline...